sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetSchemaUtils.scala

```diff
@@ -0,0 +1,574 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import java.util.Locale
+
+import scala.collection.JavaConverters._
+
+import ai.rapids.cudf.{ColumnView, DType, Table}
+import com.nvidia.spark.rapids.shims.ParquetSchemaClipShims
+import org.apache.parquet.schema._
+import org.apache.parquet.schema.Type.Repetition
+
+import org.apache.spark.sql.rapids.shims.RapidsErrorUtils
+import org.apache.spark.sql.types._
+
+object ParquetSchemaUtils extends Arm {
+  // Copied from Spark
+  private val SPARK_PARQUET_SCHEMA_NAME = "spark_schema"
+  // Copied from Spark
+  val EMPTY_MESSAGE: MessageType = Types.buildMessage().named(SPARK_PARQUET_SCHEMA_NAME)
+
+  /**
+   * Similar to Spark's ParquetReadSupport.clipParquetSchema but does NOT add fields missing
```

Comment:
Seems the comment is incorrect ?
```suggestion
   * Similar to Spark's ParquetReadSupport.clipParquetSchema but does NOT add fields that only exist
```

Reply:
Nice catch! Fixed.